etcdctl/ctlv3/command/downgrade_command.go

```diff
@@ -0,0 +1,136 @@
+		Short: "Downgrade related commands",
+	}
+
+	dc.AddCommand(NewDowngradeValidateCommand())
+	dc.AddCommand(NewDowngradeEnableCommand())
+	dc.AddCommand(NewDowngradeCancelCommand())
+
+	return dc
+}
+
+// NewDowngradeValidateCommand returns the cobra command for "downgrade validate".
+func NewDowngradeValidateCommand() *cobra.Command {
+	cc := &cobra.Command{
+		Use:   "validate <TARGET_VERSION>",
+		Short: "Validate downgrade capability before starting downgrade",
+
+		Run: downgradeValidateCommandFunc,
+	}
+	return cc
+}
+
+// NewDowngradeEnableCommand returns the cobra command for "downgrade enable".
+func NewDowngradeEnableCommand() *cobra.Command {
+	cc := &cobra.Command{
+		Use:   "enable <TARGET_VERSION>",
+		Short: "Start a downgrade action to cluster",
+
+		Run: downgradeEnableCommandFunc,
+	}
+	return cc
+}
+
+// NewDowngradeCancelCommand returns the cobra command for "downgrade cancel".
+func NewDowngradeCancelCommand() *cobra.Command {
+	cc := &cobra.Command{
+		Use:   "cancel",
+		Short: "Cancel the ongoing downgrade action to cluster",
+
+		Run: downgradeCancelCommandFunc,
+	}
+	return cc
+}
+
+// downgradeValidateCommandFunc executes the "downgrade validate" command.
+func downgradeValidateCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) < 1 {
+		cobrautl.ExitWithError(cobrautl.ExitBadArgs, errors.New("TARGET_VERSION not provided"))
+	}
+	if len(args) > 1 {
+		cobrautl.ExitWithError(cobrautl.ExitBadArgs, errors.New("too many arguments"))
+	}
+	targetVersion := args[0]
+
+	if len(targetVersion) == 0 {
+		cobrautl.ExitWithError(cobrautl.ExitBadArgs, errors.New("target version not provided"))
+	}
+
+	ctx, cancel := commandCtx(cmd)
+	cli := mustClientFromCmd(cmd)
+
+	resp, err := cli.Downgrade(ctx, clientv3.DowngradeValidate, targetVersion)
+	cancel()
+	if err != nil {
+		cobrautl.ExitWithError(cobrautl.ExitError, err)
+	}
+
+	display.DowngradeValidate(*resp)
+}
+
+// downgradeEnableCommandFunc executes the "downgrade enable" command.
+func downgradeEnableCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) < 1 {
+		cobrautl.ExitWithError(cobrautl.ExitBadArgs, errors.New("TARGET_VERSION not provided"))
+	}
+	if len(args) > 1 {
+		cobrautl.ExitWithError(cobrautl.ExitBadArgs, errors.New("too many arguments"))
+	}
+	targetVersion := args[0]
+
+	if len(targetVersion) == 0 {
+		cobrautl.ExitWithError(cobrautl.ExitBadArgs, errors.New("target version not provided"))
+	}
+
+	ctx, cancel := commandCtx(cmd)
+	cli := mustClientFromCmd(cmd)
+
+	resp, err := cli.Downgrade(ctx, clientv3.DowngradeEnable, targetVersion)
+	cancel()
+	if err != nil {
+		cobrautl.ExitWithError(cobrautl.ExitError, err)
+	}
+
+	display.DowngradeEnable(*resp)
+}
+
+// downgradeCancelCommandFunc executes the "downgrade cancel" command.
+func downgradeCancelCommandFunc(cmd *cobra.Command, args []string) {
+	ctx, cancel := commandCtx(cmd)
+	cli := mustClientFromCmd(cmd)
+
+	resp, err := cli.Downgrade(ctx, clientv3.DowngradeCancel, "")
+	cancel()
```

Comment:
This is a generic issue/question, I see that all commands just call `cancel()` directly.  Why not to use "defer cancel()" right after the corresponding line, such as line 126 in this example.